File: SAMPLES/PHP/math.php
<?php
// ============================================================
// math.php -- Fonctions, arithmetique, return pour PHP86
// ============================================================

// Fonction : carre d'un nombre
function carre($n) {
    return $n * $n;
}

// Fonction : addition
function somme($a, $b) {
    return $a + $b;
}

// Fonction : factorielle iterative
function factorielle($n) {
    $r = 1;
    $i = 1;
    while ($i <= $n) {
        $r = $r * $i;
        $i = $i + 1;
    }
    return $r;
}

// Fonction : PGCD (algorithme d'Euclide)
function pgcd($a, $b) {
    while ($b != 0) {
        $t = $b;
        $b = $a % $b;
        $a = $t;
    }
    return $a;
}

// Fonction : valeur absolue maison
function mon_abs($n) {
    if ($n < 0) {
        return 0 - $n;
    }
    return $n;
}

echo "=== math.php ===\n";

$x = 6;
$y = 4;

echo "carre(", $x, ")       = ", carre($x), "\n";
echo "somme(", $x, ", ", $y, ") = ", somme($x, $y), "\n";
echo "factorielle(5)  = ", factorielle(5), "\n";
echo "factorielle(7)  = ", factorielle(7), "\n";
echo "pgcd(48, 18)    = ", pgcd(48, 18), "\n";
echo "mon_abs(-42)    = ", mon_abs(-42), "\n";

// Builtins mathematiques
echo "abs(-99)        = ", abs(-99), "\n";
echo "max(", $x, ", ", $y, ")   = ", max($x, $y), "\n";
echo "min(", $x, ", ", $y, ")   = ", min($x, $y), "\n";

// Puissance (operateur ** ou _PHRT_POWER)
$base = 2;
$exp  = 8;
echo $base, " ** ", $exp, " = ", $base ** $exp, "\n";

echo "=== Fin math.php ===\n";

File: SAMPLES/CLOJURE/math.clj
; math.clj - defn, fn, +, -, *, /, mod, comparaisons

; --- Arithmetique de base ---
(println (+ 10 20))
(println (- 100 42))
(println (* 6 7))
(println (/ 100 5))
(println (mod 17 5))

; --- Fonctions nommees (defn) ---
(defn carre [x] (* x x))
(println (carre 7))

(defn double [n] (+ n n))
(println (double 21))

(defn somme [a b] (+ a b))
(println (somme 10 32))

; --- Fonctions multi-arg arithmetique ---
(defn moyenne [a b] (/ (+ a b) 2))
(println (moyenne 10 20))

; --- Comparaisons ---
(println (= 1 1))
(println (= 1 2))
(println (< 3 5))
(println (> 10 5))
(println (<= 5 5))
(println (>= 3 7))

; --- Predicats ---
(println (zero? 0))
(println (zero? 5))
(println (pos? 3))
(println (neg? -7))

; --- inc et dec ---
(println (inc 41))
(println (dec 43))

; --- abs et max/min ---
(defn abs-val [x] (if (neg? x) (- 0 x) x))
(println (abs-val -42))
(println (abs-val 42))

(println (max 10 20))
(println (min 10 20))
